manager/container.go

```diff
@@ -407,24 +424,22 @@ func (c *containerData) housekeeping() {
 
 	// Housekeep every second.
 	glog.V(3).Infof("Start housekeeping for container %q\n", c.info.Name)
-	lastHousekeeping := time.Now()
+	c.housekeepingTick(longHousekeeping)
 	for {
+		houseKeepingTimer := time.NewTimer(c.nextHousekeepingInterval())
 		select {
 		case <-c.stop:
 			// Stop housekeeping when signaled.
 			return
-		default:
-			// Perform housekeeping.
-			start := time.Now()
-			c.housekeepingTick()
-
-			// Log if housekeeping took too long.
-			duration := time.Since(start)
-			if duration >= longHousekeeping {
-				glog.V(3).Infof("[%s] Housekeeping took %s", c.info.Name, duration)
-			}
-		}
-
+		case finishedChan := <-c.onDemandChan:
+			c.housekeepingTick(longHousekeeping)
+			// notify the calling function that housekeeping has completed
+			close(finishedChan)
+		case <-houseKeepingTimer.C:
+			c.housekeepingTick(longHousekeeping)
```

Comment:
drain the `finishedChan` after this step. I.e.
```go
for {
  select finishedChan := <-c.onDemandChan:
    close(finishedChan)
  default:
    break
}
```

Reply:
done, although I dont think we want a for loop around it.

Reply:
why not? If there are multiple routines waiting for new stats and we just computed new stats, we should unblock them?

Reply:
done